exonum_launcher/supervisor.py

```diff
@@ -0,0 +1,108 @@
+"""Module encapsulating the interaction with the supervisor."""
+from typing import List, Optional, Any
+import json
+
+from exonum_client import ExonumClient
+from exonum_client.module_manager import ModuleManager
+
+from .configuration import Artifact, Instance
+from .instances import InstanceSpecLoader
+from .runtimes import RuntimeSpecLoader
+
+
+class Supervisor:
+    """Interface to interact with the Supervisor service."""
+
+    def __init__(self, clients: List[ExonumClient]) -> None:
+        self._clients = clients
+        self._main_client = clients[0]
+        self._loader = self._main_client.protobuf_loader()
+        self._supervisor_runtime_id: Optional[int] = None
+        self._supervisor_artifact_name: Optional[str] = None
+        self._service_module: Optional[Any] = None
+
+    def __enter__(self) -> "Supervisor":
```

Comment:
Although it is fine to use a string name of a class as a return type (the only option for Python 3.6 or below), it is possible to use the name of the class as a return type if the following module is imported:
```from __future__ import annotations```

Reply:
Yep, but until `python3.7` will become available in the default repos of common OS, I don't want to enforce users manually update `python` just to get type annotations work.

If we will use significant features assotiated with 3.7+ (e.g. use of asyncio), we will also adapt type annotations. But for now I suggest to keep things backward-compatible if possible.